src/controllers/product.controller.js

```diff
@@ -37,3 +39,42 @@ export const getAllProducts = async (req, res) => {
       .json({ error: err.message, message: 'Failed to retrieve products' });
   }
 };
+
+export const markAvailableProduct = async (req, res) => {
+  try {
+    const isAvailable = req.body.isAvailable;
+    await ProductService.markProduct({ isAvailable }, req.params.productId);
+    if(isAvailable)
+    {return res.status(200).json({ message: 'product is available now' });}
+    else{
+      return res.status(200).json({ message: 'product is not available now' });
+    }
+  } catch (error) {
+    return res.status(500).json({
+      error: error.message,
+      message: 'Failed to mark product',
+    });
+  }
```

Comment:
- We need to have a global service to update products, and that's what you'll be using to update product availability
- You're just updating, why are you checking availability?
- Write a meaningful message in the catch